common/src/TaskCompletionSource.cs

```diff
@@ -5,30 +5,22 @@
 
 namespace Microsoft.Azure.Devices.Shared
 {
-    internal sealed class TaskCompletionSource : TaskCompletionSource<int>
+    /// <summary>
+    /// A <see cref="TaskCompletionSource{boolean}"/> implementation that returns a <see cref="Task"/> when completed.
+    /// </summary>
+    /// <remarks>
+    /// This is used for .NET implementations less than .NET 5.0, which lack a native implementation of the non-generic TaskCompletionSource.
```

Comment:
It would be great to add the remarks from the TaskCompletionSource here to explain what it really is. The summary added does not really indicate what this class really represents.

https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource?view=net-5.0

Reply:
I thought it was self-explanatory, since all this class does is implement a non-generic version on TCS by setting the type as a bool. I can add the existing description for TCS into the remark, sure.